### PR TITLE
ebooks: fix deletion of holdings

### PR DESCRIPTION
**Pull request by rerowep**
Co-Authored-by: Peter Weber <peter.weber@rero.ch>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui#<xx>

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

### CODE REVIEW

File: rero_ils/modules/ebooks/utils.py [hunk 1]
```diff
@@ -137,7 +139,9 @@ def update_document_holding(record, pid):
                         source=harvested_source['source']):
                     uri_split = harvested_source['uri'].split('/')[3:]
```
Comment (jma) [inline rero_ils/modules/ebooks/utils.py:140]: Can be probably done using a replace or a regex.

File: rero_ils/modules/ebooks/utils.py [hunk 2]
```diff
@@ -125,9 +125,11 @@ def update_document_holding(record, pid):
         dbcommit=True,
         reindex=True
     )
+    source_uris = []
```
Comment (jma) [inline rero_ils/modules/ebooks/utils.py:128]: Some comments are welcome for all the following code.

**Review by jma — APPROVED** [review]

File: rero_ils/modules/ebooks/utils.py [hunk 3]
```diff
@@ -149,14 +153,13 @@ def update_document_holding(record, pid):
...
                     )
 
-    source_uris = [harvested_source.get('uri')
-                   for harvested_source in harvested_sources
-                   if 'source' in harvested_source]
-
     for holding_pid in Holding.get_holdings_pid_by_document_pid(pid):
         holding = Holding.get_record_by_pid(holding_pid)
         for electronic_location in holding.get('electronic_location', []):
             if electronic_location.get('source') and \
```
Comment (PascalRepond) [inline rero_ils/modules/ebooks/utils.py:159]: When updating, we should also delete holdings if their location is not `online` anymore.

**Review by PascalRepond — CHANGES_REQUESTED** [review]

**Review by lauren-d — APPROVED** [review]

**Review by Garfield-fr — APPROVED** [review]

**Review by PascalRepond — APPROVED** [review]

File: rero_ils/modules/ebooks/utils.py (hunk 3 above)
Comment (rerowep) [reply rero_ils/modules/ebooks/utils.py:159]: Should be done during update of the library.

**Pull request merged**